From 75c74cec42b0a17e86e96b8c9cd0164b18891c0f Mon Sep 17 00:00:00 2001
From: bitcoffee <liuxin350@huawei.com>
Date: Wed, 15 Feb 2023 20:11:56 +0800
Subject: [PATCH 1/4] net, ipv4: A new bit is added to indicate whether to
 delay link establishment using bpf 

hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I6CPF6

---

The bpf_defer_connect bit is added for inet_sock to indicate 
whether the current socket is changed to the bpf program to 
delay link establishment.

Signed-off-by: bitcoffee <liuxin350@huawei.com>
---
 include/net/inet_sock.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/net/inet_sock.h b/include/net/inet_sock.h
index bf5654ce711e..497987bde8ad 100644
--- a/include/net/inet_sock.h
+++ b/include/net/inet_sock.h
@@ -238,10 +238,15 @@ struct inet_sock {
 				nodefrag:1;
 	__u8			bind_address_no_port:1,
 				recverr_rfc4884:1,
-				defer_connect:1; /* Indicates that fastopen_connect is set
+				defer_connect:1, /* Indicates that fastopen_connect is set
 						  * and cookie exists so we defer connect
 						  * until first data frame is written
 						  */
+				bpf_defer_connect:1; /* specifies the ebpf program for
+						      * traffic orchestration so we defer
+						      * connect until first data frame is
+						      * written
+						      */
 	__u8			rcv_tos;
 	__u8			convert_csum;
 	int			uc_index;
-- 
2.33.0

